pkg/diffs/diffs_test.go

```diff
@@ -148,3 +154,174 @@ func makeConfig(p []prowconfig.Presubmit) *prowconfig.Config {
+							Image:   "setup-image",
+							Command: []string{"bash-script"},
+						},
+						{
+							Name:    "test",
+							Image:   "test-image",
+							Command: []string{"bash-script"},
+						},
+						{
+							Image:   "teardown-image",
+							Name:    "teardown",
+							Command: []string{"bash-script"},
+						},
+					},
+					Volumes: []v1.Volume{
+						{
+							Name: "test-volume",
+							VolumeSource: v1.VolumeSource{
+								EmptyDir: &v1.EmptyDirVolumeSource{}},
+						},
+					},
+				},
+			},
+		},
+	}
+	testCases := []struct {
+		name         string
+		getTemplates func() (map[string]*templateapi.Template, map[string]*templateapi.Template)
+		expected     map[string]*templateapi.Template
+	}{
+		{
+
+			name: "no changes",
+			getTemplates: func() (map[string]*templateapi.Template, map[string]*templateapi.Template) {
+				templates := makeBaseTemplates(baseParameters, baseObjects)
+				return templates, templates
+			},
+			expected: map[string]*templateapi.Template{},
+		},
+		{
+			name: "add new template",
+			getTemplates: func() (map[string]*templateapi.Template, map[string]*templateapi.Template) {
+				prTemplates := makeBaseTemplates(baseParameters, baseObjects)
+
+				prTemplates["templateNEW"] = &templateapi.Template{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "templateNEW",
+						Annotations: map[string]string{
+							"description": "description",
+						},
+					},
+					Parameters: baseParameters,
+					Objects:    baseObjects,
+				}
+				return makeBaseTemplates(baseParameters, baseObjects), prTemplates
+			},
+			expected: map[string]*templateapi.Template{
+				"templateNEW": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "templateNEW",
+						Annotations: map[string]string{
+							"description": "description",
+						},
+					},
+					Parameters: baseParameters,
+					Objects:    baseObjects,
+				},
+			},
+		},
+		{
+			name: "change existing template",
+			getTemplates: func() (map[string]*templateapi.Template, map[string]*templateapi.Template) {
+				prTemplates := makeBaseTemplates(baseParameters, baseObjects)
+
+				prTemplates["template1"].Parameters = append(prTemplates["template1"].Parameters,
+					templateapi.Parameter{
+						Name:  "NEW_PARAM",
+						Value: "TEST_VALUE",
+					})
+				return makeBaseTemplates(baseParameters, baseObjects), prTemplates
+			},
+			expected: map[string]*templateapi.Template{
+				"template1": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "template1",
+						Annotations: map[string]string{
+							"description": "description",
+						},
+					},
+					Parameters: append(baseParameters, templateapi.Parameter{
+						Name:  "NEW_PARAM",
+						Value: "TEST_VALUE",
+					}),
+					Objects: baseObjects,
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			before, after := testCase.getTemplates()
+			changedTemplates := GetChangedTemplates(before, after, logrus.NewEntry(logrus.New()))
```

Comment:
Can add test name as a `WithField` for better logging debug too

Reply:
addressed